pkg/resourceinterpreter/interpreter.go

```diff
@@ -95,6 +98,25 @@ func (i *customResourceInterpreterImpl) GetReplicas(object *unstructured.Unstruc
 	return
 }
 
+// ReviseReplica revises the replica of the given object.
+func (i *customResourceInterpreterImpl) ReviseReplica(object *unstructured.Unstructured, replica int64) (*unstructured.Unstructured, error) {
+	klog.V(4).Infof("Begin to revise replicas for object: %v %s/%s.", object.GroupVersionKind(), object.GetNamespace(), object.GetName())
+
+	patch, patchType, hookEnabled, err := i.customizedInterpreter.Retain(context.TODO(), &webhook.RequestAttributes{
```

Comment:
Why use `customizedInterpreter.Retain` here? Do you want to share the same logic?

Reply:
Actually the logic is the same, I could add a new `customizedInterpreter.ReviseReplica` method but it will be just the same as `customizedInterpreter.Retain`

Reply:
Maybe we can rename the `customizedInterpreter.Retain` to `Patch`, `ReviseReplica` and `Retain` can share this common method.

Reply:
Not rename, we can abstract a common method shared by `Retain()` and `ReviseReplica()`, for code readability.